main/cmds.go

```diff
@@ -82,24 +86,39 @@ func enablePre(ctx *log.Context, seqNum int) error {
 	return nil
 }
 
-func enable(ctx *log.Context, h vmextension.HandlerEnvironment, seqNum int) error {
+func enable(ctx *log.Context, h vmextension.HandlerEnvironment, seqNum int) (string, error) {
 	// parse the extension handler settings (not available prior to 'enable')
 	cfg, err := parseAndValidateSettings(ctx, h.HandlerEnvironment.ConfigFolder)
 	if err != nil {
-		return errors.Wrap(err, "failed to get configuration")
+		return "", errors.Wrap(err, "failed to get configuration")
 	}
 
 	dir := filepath.Join(dataDir, downloadDir, fmt.Sprintf("%d", seqNum))
 	if err := downloadFiles(ctx, dir, cfg); err != nil {
-		return errors.Wrap(err, "processing file downloads failed")
+		return "", errors.Wrap(err, "processing file downloads failed")
 	}
 
-	if err := runCmd(ctx, dir, cfg); err != nil {
-		return err
+	// execute the command, save its error
+	runErr := runCmd(ctx, dir, cfg)
+
+	// collect the logs if available
+	stdoutF, stderrF := logPaths(dir)
+	stdoutTail, err := tailFile(stdoutF, maxTailLen)
+	if err != nil {
+		ctx.Log("message", "error tailing stdout logs", "error", err)
+	}
+	stderrTail, err := tailFile(stderrF, maxTailLen)
+	if err != nil {
+		ctx.Log("message", "error tailing stderr logs", "error", err)
 	}
+	msg := fmt.Sprintf("\n[stdout]\n%s\n[stderr]\n%s", string(stdoutTail), string(stderrTail))
```

Comment:
There might be people relying on that in other places but I just did not see any of those in ARM templates. ARM templates mostly rely on splitting by \n. I feel like this would also be a good reminder that they shouldn't rely on undocumented features when users get broken downstream and have to fix it?